### ISSUE TITLE
Not fully working within ts and tsx files

### ISSUE THREAD

**Issue by icaroponce** [ap/vim-css-color#118]
Hexadecimals aren't always properly highlighted on `.ts`  and `.tsx` files. When the hexadecimal is within a comment, it's highlighted though. 

![2020-02-11T11:27:13](https://user-images.githubusercontent.com/8094399/74228837-8a3eec00-4cc1-11ea-88e3-090ed7749f44.png)

**Comment by ap** [comment]
Works for me using [leafgarland/typescript-vim](https://github.com/leafgarland/typescript-vim)… do you use something else?

**Comment by icaroponce** [comment]
@ap currently I don't have any specific TS syntax plugin installed. Only `pangloss/vim-javascript`.

**Comment by ap** [comment]
That one [has no detection for TypeScript files](/pangloss/vim-javascript/blob/3c90d0cc37bb8b78422f647e62587f498a5dd7bd/ftdetect/javascript.vim), so the question is where the highlighting comes from. What does `:set syntax?` say in your TypeScript buffers? And if it’s some TypeScript-specific value, do you know where that’s coming from?

**Comment by icaroponce** [comment]
`:set syntax` displays `syntax=typescript`. I don't really see any other plugin or configuration that could be setting the value.
In case it helps, more info about my neovim: 

```
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.3/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
```

I can also provide the full list of plugins currently installed or even my whole `.vimrc` if needed.

**Comment by ap** [comment]
[Looking around your dotfiles](/icaroponce/dotfiles/search?q=typescript), [it seems like](/HerringtonDarkholme/yats.vim/blob/4d76c5391b7975bd6dd09c0562c0fd4d78e14fa0/ftdetect/typescript.vim) that comes from [HerringtonDarkholme/yats.vim](/HerringtonDarkholme/yats.vim)? With that installed I get the same result that you do, so let’s confirm: does applying the following patch to CSS Color change anything for you?

````diff
diff --git i/after/syntax/typescript.vim w/after/syntax/typescript.vim
--- i/after/syntax/typescript.vim
+++ w/after/syntax/typescript.vim
@@ -1,6 +1,7 @@
 " https://github.com/leafgarland/typescript-vim
 " should already be including the xml syntax
 call css_color#init('hex', 'extended'
- \, 'typescriptComment,typescriptLineComment,typescriptStringS,typescriptStringD,'
- \. 'typescriptStringB,typescriptCommentSkip,typescriptDocComment'
+ \, 'typescriptComment,typescriptLineComment,'
+ \. 'typescriptStringS,typescriptStringD,typescriptStringB,typescriptString,typescriptStringProperty,'
+ \. 'typescriptCommentSkip,typescriptDocComment'
  \)
````

**Comment by icaroponce** [comment]
Unfortunately my dotfiles repo isn't up-to-date. I don't use the `yats.vim` anymore due to weird behavior when working with JSX. Anyways, I applied the patch you suggested and it solved the problem, so thanks for the efforts.

**Comment by ap** [comment]
Well, this is kinda frustrating. If you aren’t using yats.vim then why does the patch work? I’d like to figure that out so I’m not just randomly patching things until they seem to work for someone. The value is getting set in your setup, so *something* you use *is* setting it – so what is that? It can’t be impossible to find out.

Does `:verbose au BufRead *.ts` give any useful hint??

**Comment by icaroponce** [comment]
```
:verbose au BufRead *.ts                                                              
--- Autocommands ---                                                                  
PrettierFileDetect  BufReadPost                                                       
    *.ts      setfiletype typescript                                                  
        Last set from ~/.vim/plugged/vim-prettier/ftdetect/typescript.vim line 2      
filetypedetect  BufReadPost                                                           
    *.ts      setf typescript                                                         
        Last set from /usr/share/nvim/runtime/filetype.vim line 1712                  
    *.ts      set ft=typescript                                                       
        Last set from ~/.vim/plugged/vim-polyglot/ftdetect/polyglot.vim line 649
```

**Comment by ap** [comment]
Ah. So what really happened isn’t that stopped using yats.vim, you just switched from downloading yats.vim explicitly to relying on the copy of yats.vim that Vim and NeoVim now ship built-in. 🙂 And the reason I couldn’t reproduce your bug was because my Vim was a little older and didn’t include the upstreamed yats.vim.

**Comment by ap** [comment]
Fixed in d2ccb376ebfeeef842456a7883164ef2c0d1656c. Share and enjoy 🙂
